src/test/java/core/basesyntax/service/impl/TransactionExecutorTest.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.impl.BalanceHandler;
+import core.basesyntax.strategy.impl.PurchaseHandler;
+import core.basesyntax.strategy.impl.ReturnHandler;
+import core.basesyntax.strategy.impl.SupplyHandler;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+public class TransactionExecutorTest {
+    private static final String DEFAULT_VALID_FILE = "src/main/resources/example.csv";
+    private FruitTransactionParserImpl fruitTransactionParser = new FruitTransactionParserImpl();
+    private FileReaderCsv fileReaderCsv = new FileReaderCsv();
+
+    @Test
+    void execute_validData_Ok() {
+        Map<FruitTransaction.Operation, OperationHandler> strategyMap = new HashMap<>();
+        strategyMap.put(FruitTransaction.Operation.BALANCE, new BalanceHandler());
+        strategyMap.put(FruitTransaction.Operation.PURCHASE, new PurchaseHandler());
+        strategyMap.put(FruitTransaction.Operation.RETURN, new ReturnHandler());
+        strategyMap.put(FruitTransaction.Operation.SUPPLY, new SupplyHandler());
+        TransactionExecutor transactionExecutor = new TransactionExecutor(strategyMap);
+        transactionExecutor.execute(fruitTransactionParser.parse(fileReaderCsv
+                .read(DEFAULT_VALID_FILE)));
+        assertEquals(transactionExecutor.getStrategy()
+                .get(FruitTransaction.Operation.BALANCE).getClass(), BalanceHandler.class);
```

Comment:
this assert is redundant and does not have anything to do with the case of testing how `execute` method works, and in general this `getStrategy()` method is redundant, I'd say we never should have a situation where we need to get the service of some other service.
Instead, we should assert storage data here, or write a report and check its file contents if you really want to commit to this end-to-end test